### ISSUE TITLE
replace `assert` with `ALPAKA_ASSERT_ACC` in device code

### ISSUE THREAD

**Issue by fwyzard** [cms-sw/cmssw#44792]
Plenty of files use `assert` in alpaka device code.
This currently works, but not all Alpaka backends (_e.g._ SYCL) support `assert` for all devices.

The best approach would be to replace `assert(...)` with `ALPAKA_ASSERT_ACC(...)`.

`ALPAKA_ASSERT_ACC` compiles to `assert` on the backends that support it, and to nothing on backends that do not support `assert`.

It can also be centrally disabled at compile time if its impact on the performance turns out to be significant.

A (possibly incomplete) of alpaka-related files that uses `assert` is:
### Core
```
HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h
HeterogeneousCore/AlpakaInterface/test/alpaka/testAtomicPairCounter.dev.cc
HeterogeneousCore/AlpakaInterface/test/alpaka/testOneRadixSort.dev.cc
HeterogeneousCore/AlpakaInterface/test/alpaka/testRadixSort.dev.cc
HeterogeneousCore/AlpakaInterface/test/alpaka/testSimpleVector.dev.cc
HeterogeneousCore/AlpakaInterface/test/alpaka/testWorkDivision.dev.cc
HeterogeneousCore/AlpakaInterface/test/alpaka/testBuffer.dev.cc
```

### Pixel
```
CondFormats/SiPixelObjects/interface/alpaka/SiPixelGainCalibrationForHLTUtilities.h
DataFormats/SiPixelClusterSoA/test/alpaka/Clusters_test.dev.cc
DataFormats/SiPixelDigiSoA/test/alpaka/DigiErrors_test.dev.cc
DataFormats/SiPixelDigiSoA/test/alpaka/Digis_test.dev.cc
DataFormats/TrackingRecHitSoA/interface/alpaka/TrackingRecHitsSoACollection.h
DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.dev.cc
RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc
RecoLocalTracker/SiPixelRecHits/plugins/alpaka/PixelRecHits.h
RecoLocalTracker/SiPixelRecHits/plugins/alpaka/PixelRecHitKernels.dev.cc
RecoTracker/PixelSeeding/plugins/alpaka/CACell.h
RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernels.dev.cc
RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernels.h
RecoTracker/PixelSeeding/plugins/alpaka/CAPixelDoubletsAlgos.h
RecoTracker/PixelSeeding/plugins/alpaka/RiemannFit.dev.cc
```

### ECAL
```
RecoLocalCalo/EcalRecProducers/plugins/alpaka/TimeComputationKernels.h
```

(some of these may be false positives, _e.g._ `assert` used in host code).

**Comment by fwyzard** [comment]
assign heterogeneous

**Comment by fwyzard** [comment]
type tracker, tracking, pf

**Comment by fwyzard** [comment]
type ecal

**Comment by fwyzard** [comment]
type tracking

**Comment by fwyzard** [comment]
type trk

**Comment by fwyzard** [comment]
@AdrianoDee @thomreis FYI

**Comment by makortel** [comment]
> HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h

The `assert()` in
https://github.com/cms-sw/cmssw/blob/3fb8b0edb88bcba34579ca4b9867427862eda51d/HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h#L76
should be called from host code only (by ESProducer constructor), and therefore I think should be kept as `assert()`.

**Comment by fwyzard** [comment]
I've attempted to migrate all device-side `assert()` to `ALPAKA_ASSERT_ACC()` in https://github.com/cms-sw/cmssw/pull/44802 .

**Comment by fwyzard** [comment]
Fixed by #44802.
